src/crimson/net/SocketConnection.cc

```diff
@@ -914,30 +895,28 @@ SocketConnection::execute_open()
   seastar::now().forward_to(std::move(h.promise));
   seastar::with_gate(pending_dispatch, [this] {
       // start background processing of tags
-      read_tags_until_next_message();
-      return seastar::keep_doing([this] {
-          return read_message()
-            .then([this] (MessageRef msg) {
-              // start dispatch, ignoring exceptions from the application layer
-              seastar::with_gate(pending_dispatch, [this, msg = std::move(msg)] {
-                  return dispatcher.ms_dispatch(this, std::move(msg))
-                    .handle_exception([] (std::exception_ptr eptr) {});
-                });
-              // return immediately to start on the next message
-              return seastar::now();
-            });
-        }).handle_exception_type([this] (const std::system_error& e) {
+      return handle_tags()
+        .handle_exception_type([this] (const std::system_error& e) {
           if (e.code() == error::connection_aborted ||
               e.code() == error::connection_reset) {
-            return dispatcher.ms_handle_reset(this);
+            logger().warn("{} open fault: {}", *this, e);
```

Comment:
can we extract these `logger().warn("{} open fault: {}", *this, e);` calls out before the `if` clause. seems we are going to print it no matter what `e` is.

Reply:
done.